VideoWeb/VideoWeb/ClientApp/src/app/services/user-media.service.ts

```diff
@@ -13,16 +13,19 @@ export class UserMediaService {
 
     private readonly preferredCamCache: SessionStorage<UserMediaDevice>;
     private readonly preferredMicCache: SessionStorage<UserMediaDevice>;
+    private readonly showDialogChooseDevicesOnInit: SessionStorage<boolean>;
+
     readonly PREFERRED_CAMERA_KEY = 'vh.preferred.camera';
     readonly PREFERRED_MICROPHONE_KEY = 'vh.preferred.microphone';
-
+    readonly CHOOSE_DEVICES_ON_INIT_IN_WR_KEY = 'vh.first.time.in.waitingroom';
     availableDeviceList: UserMediaDevice[];
 
     connectedDevices: BehaviorSubject<UserMediaDevice[]> = new BehaviorSubject([]);
 
     constructor(private logger: Logger) {
         this.preferredCamCache = new SessionStorage(this.PREFERRED_CAMERA_KEY);
         this.preferredMicCache = new SessionStorage(this.PREFERRED_MICROPHONE_KEY);
+        this.showDialogChooseDevicesOnInit = new SessionStorage(this.CHOOSE_DEVICES_ON_INIT_IN_WR_KEY);
```

Comment:
The user media service should not care whether or not the user has visited the WR for the first time. Tracking this hear takes away from the service's responsibility of only managing devices

Reply:
moved to waiting room (base class)